library/headers.inc.php

```diff
@@ -42,43 +46,66 @@ function include_css_library($path)
-    <?php 
-    } }
+    /* First checking for any library in a directory matching the string "/jquery-min-/".
+     * When one is found, use that value in the URL string, then add "index.js"
+     * In order to decrease the chance of including and library calling problems, some measures were taken:
+     * - The array in the argument is completely converted to lower-case, meaning that there is not any sensitivity between jQuery & jquery or fancyBox & fancybox
+     * - Some libraries can be called with or without "-". For example: "font-awesome" and "fontawesome" => those 2 are the same thing
+     */
+    
+    $library_array = array_map('strtolower', $library_array);
+
+    foreach ($library_array as $v){ 
+        if (preg_match("/jquery-min-/", $v)) {?>
+            <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path'].$v ; ?>/index.js"></script>    
+        <?php 
+        }
+    }   
         
-    if (in_array("bootstrap",$library_array)){   ?>
+    if (in_array("bootstrap", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['standard_js_path']; ?>bootstrap-3-3-4/dist/css/bootstrap.min.css" type="text/css">
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>bootstrap-3-3-4/dist/js/bootstrap.min.js"></script>
     <?php
     }
-    if (in_array("fancybox",$library_array)){   ?>
+
+    if (in_array("fancybox", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>fancybox/jquery.fancybox-1.2.6.css" media="screen" />
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>fancybox/jquery.fancybox-1.2.6.js"></script>
     <?php
     }
-    if (in_array("knockout",$library_array)){   ?>
+
+    if (in_array("knockout", $library_array)) {   ?>
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>knockout/knockout-3.4.0.js"></script>
     <?php
     }
-    if (in_array("datepicker",$library_array)){   ?>
+
+    if (in_array("datepicker", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>jquery-datetimepicker/jquery.datetimepicker.css" media="screen" />
         <script type="text/javascript" src="<?php echo $GLOBALS['standard_js_path']; ?>jquery-datetimepicker/jquery.datetimepicker.full.min.js"></script>
     <?php
     }
-    if (in_array('font-awesome', $library_array)) {?>
+
+    if (in_array('font-awesome', $library_array) || in_array('fontawesome', $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['assets'] ?>/fonts/font-awesome-4-6-3/css/font-awesome.min.css" type="text/css">
     <?php
     }
-    if (in_array("jquery-ui",$library_array)){   ?>
+
+    if (in_array("jquery-ui", $library_array) || in_array("jqueryui", $library_array)) {   ?>
         <link rel="stylesheet" href="<?php echo $GLOBALS['css_path']; ?>jquery-ui-1-12-1/jquery-ui.css" media="screen" />
```

Comment:
Do you think this needs to be made to work like the check for jquery. We at some point would have more that one jquery-ui

Reply:
Agree.  The jquery thing was the best stupid idea I have had in a long time.

Reply:
Maybe, but, why not having the latest version of jQuery? I mean, they won't probably deprecate anything just because they want... Anyway, I think that there should be the option to call only "jquery" and that would get the latest available version of jquery in the local system. I mean, sometimes, we may not need to really specify a version.

Reply:
We have tried that already.
When we have killed every legacy use of JQuery in the system, then we will consider it.  In the meantime, we can always have the plain old JQuery.js file be the latest and greatest.  Incremental changes and granular control are vital when doing "stranglevine" refactors.  That is a method of code refactoring where you gradually replace the old with the new.  It is hard, but you break less things, and you can do it without a huge budget.